Client/src/Components/Admin/AdminReviewView.tsx

```diff
@@ -0,0 +1,113 @@
+import { Box, Button, Grid, TextField, Typography, createMuiTheme, withStyles } from '@material-ui/core'
+import { IApprovedDatasetModel, IFlaggedDatasetQuery } from '../../Models/Datasets/IApprovedDatasetModel'
+import React, { useEffect, useState } from 'react'
+import { adminApprovedDataset, callRejectDataset, flagDataset } from '../../Remote/Endpoints/DatasetEndpoints'
+
+import { AdminReviewList } from './AdminReviewList'
+import { DatasetUploadForm } from '../DatasetUpload/DatasetUploadForm'
+
+export function AdminReviewView() {
+
+    const [datasetState, setDatasetState] = useState([])
+    const [editable, setEditable] = useState(false)
+    const [dataset, setDataset] = useState<IApprovedDatasetModel>()
+    const [comment, setComment] = useState("")
+    const [flaggedComment, setFlaggedComment] = useState("")
+    const [update, setUpdate] = useState(0)
+
+    const handleDeleteDataset = async () => {
+        await callRejectDataset(dataset.id)
+        setUpdate(update + 1)
+        handleDatasetChange(null)
+    }
+
+    const handleEditDataset = () => {
+        setEditable(!editable)
+    }
+
+    const handleFlagDataset = async () => {
+        const query: IFlaggedDatasetQuery = { datasetId: dataset.id, flaggedComments: flaggedComment, additionalComments: comment }
+        await flagDataset(query)
+    }
+
+    const handleApproveDataset = async (datasetId: IApprovedDatasetModel) => {
+        const query: IFlaggedDatasetQuery = { datasetId: dataset.id, additionalComments: comment }
+        await adminApprovedDataset(query)
+        setUpdate(update + 1)
+        handleDatasetChange(null)
+    }
+
+    const handleDatasetChange = (newDataset: IApprovedDatasetModel) => {
+        setDataset(newDataset)
+    }
+
+    const handleCommentChange = (event) => {
+        setComment(event.target.value)
+    }
+
+    const handleFlaggedCommentChange = (event) => {
+        setFlaggedComment(event.target.value)
+    }
+    return (
+
+        <Grid justify="flex-start" container spacing={4}>
+            <Grid xs={3}>
+                <Box p={4} pt={7}>
+                    <Typography variant='h5'>
+                        Datasets to be reviewed
+                    </Typography>
+                    <br></br>
+                    <AdminReviewList
+                        datasets={datasetState}
+                        onChange={handleDatasetChange}
+                        update={update}
+                    />
+                    <br></br>
+                    <Button id="toggle-edit" onClick={handleEditDataset} color="primary" variant="contained">Edit</Button>&nbsp;&nbsp;&nbsp;
```

Comment:
Are we able to disable the other buttons while in edit mode? Also, i still think that updating a dataset does not need approval. Admin should be able to edit out a typo and still be able to flag it. We should ask Kiyanda for his feedback on monday for this.